File: backend/src/main/kotlin/com/example/backend/models/FactureElectronique.kt
package com.example.backend.models

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table
import org.hibernate.annotations.ColumnDefault

@Entity
@Table(name = "facture_electronique")
class FactureElectronique {
  @Id
  @Column(name = "id", nullable = false)
  var id: Int? = null

  @Column(name = "facturation_id")
  var facturationId: Long? = null

  @Column(name = "numeroTelephone", length = 15)
  var numeroTelephone: String? = null

  @Column(name = "montant")
  var montant: Int? = null

  @ColumnDefault("0")
  @Column(name = "supprimer")
  var supprimer: Int? = null
}

File: backend_py/pharmaPython/app/models/facture_electronique.py
# models/facture_electronique.py
from sqlalchemy import Column, Integer, BigInteger, String, ForeignKey
from sqlalchemy.orm import declarative_base, relationship

from app.core.db import Base

class FactureElectronique(Base):
  __tablename__ = "facture_electronique"
  __table_args__ = {"extend_existing": True}

  id = Column(Integer, primary_key=True)
  facturation_id = Column(Integer, ForeignKey("facturation.id"), nullable=True)
  numero_telephone = Column(String(15), nullable=True)
  montant = Column(Integer, nullable=True)
  supprimer = Column(Integer, nullable=True, server_default="0", default=0)

  facturation = relationship("Facturation")

  # def __repr__(self) -> str:
  #   return (
  #     f"<FactureElectronique(id={self.id}, facturation_id={self.facturation_id}, "
  #     f"numero_telephone={self.numero_telephone!r}, montant={self.montant}, supprimer={self.supprimer})>"
  #   )
